### PR TITLE
Add an additional plugin repository needed in the build and additional dependency ignores.

### PR DESCRIPTION
**Pull request by cunningt**
Add some ignoreUsedDependencies that are needed in the build, and add a plugin repository statement for SNAPSHOTS because the syndesis-build-helper-maven-plugin is used and I'm getting errors when running mvn clean and it can't resolve the plugin.

### CODE REVIEW

File: app/rest/runtime/pom.xml [hunk 1]
```diff
@@ -60,6 +60,18 @@
       <url>https://jcenter.bintray.com</url>
     </pluginRepository>
 
+    <pluginRepository>
+      <id>sonatype-snapshot</id>
+      <name>Sonatype snapshot repository</name>
+      <url>https://oss.sonatype.org/service/local/repositories/snapshots/content</url>
+      <releases>
+        <enabled>true</enabled>
```
Comment (rhuss) [inline app/rest/runtime/pom.xml]: Shouldn't be set this to `false` ?

**Comment by rhuss** [comment]
@zregvart @cunningt actually I wonder why this does not bite us upstream. happy with the additional repo.

**Comment by zregvart** [comment]
Is this something we still need to address?

**Comment by rhuss** [comment]
@cunningt sorry, I forgot the original rationale for this PR. Is this still required, and if so, could you please elaborate a bit on the issues ? Thanks !

**Pull request closed without merging**